R/linter_tags.R

```diff
@@ -0,0 +1,186 @@
+#' }
+#'
+#' Tags should be snake_case.
+#'
+#' @return
+#' A data frame with columns 'linter', 'package' and 'tags':
+#'
+#' \describe{
+#' \item{linter}{A character column naming the function associated to the linter.}
+#' \item{package}{A characetr column containing the name of the package providing the linter.}
+#' \item{tags}{A list column containing tags associated to the linter.}
+#' }
+#'
+#' @examples
+#' lintr_linters <- available_linters()
+#'
+#' # If the package doesn't exist or isn't installed, an empty data frame will be returned
+#' available_linters("does-not-exist")
+#'
+#' lintr_linters2 <- available_linters(c("lintr", "does-not-exist"))
+#' identical(lintr_linters, lintr_linters2)
+#' @export
+available_linters <- function(packages = "lintr") {
+  if (!is.character(packages)) {
+    stop("`packages` must be a character vector.")
+  }
+
+  # Handle multiple packages
+  if (length(packages) > 1L) {
+    return(do.call(rbind, lapply(packages, available_linters)))
+  }
+
+  csv_file <- system.file("lintr", "linters.csv", package = packages)
+
+  # Make sure we always return a valid data frame
+  # data.frame(...) and as.data.frame(...) don't handle zero-row list-columns properly, so we need to manually create
+  # the structure.
+  empty_linters <- structure(list(
+    linter = character(),
+    package = character(),
+    tags = list()
+  ), row.names = integer(0), class = "data.frame")
+
+  if (!file.exists(csv_file)) {
+    return(empty_linters)
+  }
+  available <- utils::read.csv(csv_file, encoding = "UTF-8", as.is = TRUE)
+
+  # Check that the csv file contains two character columns, named 'linter' and 'tags'.
+  # Otherwise, fallback to an empty data frame.
+  if (!ncol(available) == 2L) {
+    warning(
+      "`linters.csv` must contain two columns, 'linter' and 'tags'.\nPackage '",
+      packages, "' contains ", ncol(available), " columns instead."
+    )
+    return(empty_linters)
+  } else if (!identical(colnames(available), c("linter", "tags"))) {
+    warning(
+      "`linters.csv` must contain two columns, 'linter' and 'tags'.\nPackage '",
+      packages, "' contains columns '", colnames(available)[1L], "' and '", colnames(available)[2L], "' instead."
+    )
+    return(empty_linters)
+  } else if (nrow(available) == 0L) {
+    # Circumvent empty list problem in data.frame(...)
+    return(empty_linters)
+  }
+
+  structure(list(
+    linter = available[["linter"]],
+    package = rep_len(packages, nrow(available)),
+    tags = strsplit(available[["tags"]], split = " ", fixed = TRUE)
+  ), row.names = c(NA, -nrow(available)), class = "data.frame")
+}
+
+#' Generate Rd fragment for the Tags section of a linter
+#'
+#' @param linter_name Name of the linter to generate Rd code for.
+#'
+#' @noRd
+rd_tags <- function(linter_name) {
+  linters <- available_linters()
+  tags <- sort(linters[["tags"]][[match(linter_name, linters[["linter"]])]])
+
+  c(
+    "\\section{Tags}{",
+    if (length(tags)) {
+      paste(
```

Comment:
i prefer using `toString()` since it's there already

Reply:
changed